tests/unit/test_charm.py

```diff
@@ -724,3 +724,145 @@ def test_get_rgw_endpoints_action_after_traefik_is_integrated(
         }
         action_event.set_results.assert_called_with(expected_endpoints)
         action_event.fail.assert_not_called()
+
+    @patch("microceph.subprocess.run")
+    def test_enter_maintenance_action_succeed(self, subprocess_run):
+        output = "plan or error message"
+        mock_process = MagicMock()
+        mock_process.stdout = PropertyMock(return_value=output)
+
+        subprocess_run.return_value = mock_process
+        action_event = MagicMock()
+        name = "node-1"
+        force = False
+        dry_run = False
+        set_noout = True
+        stop_osds = False
+        action_event.params = {
+            "name": name,
+            "set-noout": set_noout,
+            "stop-osds": stop_osds,
+            "dry-run": dry_run,
+            "force": force,
+        }
+        cmd = [
+            "sudo",
+            "microceph",
+            "cluster",
+            "maintenance",
+            "enter",
+            "node-1",
+            f"--set-noout={set_noout}",
+            f"--stop-osds={stop_osds}",
+            f"--force={force}",
+            f"--dry-run={dry_run}",
```

Comment:
Please consider making these kind of tests more declarative by reducing the use of variables. Otherwise the logic in the tests starts to resemble the logic in the function, and it's easy to miss things.

Reply:
This is not relevant anymore in the latest commits